pytorch_lightning/strategies/deepspeed.py

```diff
@@ -736,6 +736,11 @@ def save_checkpoint(self, checkpoint: Dict, filepath: _PATH) -> None:
             checkpoint: The checkpoint state dictionary
             filepath: write-target file's path
         """
+        # since deepspeed creates subfolders instead of checkpoint files at the filepath
+        # it's better to remove the filepath completely before saving a new one to avoid
+        # creating multiple sub-folders (more info: #11687)
+        self.remove_checkpoint(filepath)
```

Comment:
another patch here can be at L-756
```
self.deepspeed_engine.save_checkpoint(filepath, client_state=checkpoint, tag='some_static_string')
```
so that deepspeed doesn't use dynamic tags to generate multiple sub-folders rather replace it.

Reply:
update tag='checkpoint' instead since `tag=''` saves the sharded weights in the folder itself where 2 more files are present.
so now it will be `["latest", "zero_to_fp32.py", "checkpoint"]`.